src/pages/New.js

```diff
@@ -25,20 +25,23 @@ export default function New() {
     postRestaurant({
       imgSrc:
         "https://static.lieferando.de/images/restaurants/de/OPROO0P/logo_465x320.png",
-      title: "Pasta Pronto",
+      title: input,
       categories: ["pizza", "salad", "pasta"],
       distance: 6,
       rating: 3.1,
       description: "Eat italian"
     });
   }
-
+  const [input, setTitleInput] = useState();
   return (
     <Modal hideBackdrop onAccept={handleAccept}>
       <AddCircleIcon />
       <ModalTitle>Add Restaurant</ModalTitle>
       <ModalSection>Name</ModalSection>
-      <TextInput placeholder="Enter restaurant name" />
+      <TextInput
+        placeholder="Enter restaurant name"
+        onChange={event => setTitleInput(event.target.value)}
```

Comment:
looks good!
If you use an `onChange` it would make sense to set `value` too to control the component.
See https://reactjs.org/docs/forms.html#controlled-components